plugins/aws/resoto_plugin_aws/resource/ecr.py

```diff
@@ -0,0 +1,73 @@
+    kms_key: Optional[str] = field(default=None, metadata={"description": "If you use the KMS encryption type, specify the KMS key to use for encryption. The alias, key ID, or full ARN of the KMS key can be specified. The key must exist in the same Region as the repository. If no key is specified, the default Amazon Web Services managed KMS key for Amazon ECR will be used."})  # fmt: skip
+
+
+@define(eq=False, slots=False)
+class AwsEcrRepository(AwsResource):
+    kind: ClassVar[str] = "aws_ecr_repository"
+    api_spec: ClassVar[AwsApiSpec] = AwsApiSpec("ecr", "describe-repositories", "repositories")
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("repositoryName"),
+        "tags": S("Tags", default=[]) >> ToDict(),
+        "name": S("repositoryName"),
+        "ctime": S("createdAt"),
+        "repository_arn": S("repositoryArn"),
+        "registry_id": S("registryId"),
+        "repository_uri": S("repositoryUri"),
+        "image_tag_mutability": S("imageTagMutability"),
+        "image_scan_on_push": S("imageScanningConfiguration", "scanOnPush"),
+        "encryption_configuration": S("encryptionConfiguration") >> Bend(AwsEcrEncryptionConfiguration.mapping),
+    }
+    repository_arn: Optional[str] = field(default=None, metadata={"description": "The Amazon Resource Name (ARN) that identifies the repository. The ARN contains the arn:aws:ecr namespace, followed by the region of the repository, Amazon Web Services account ID of the repository owner, repository namespace, and repository name. For example, arn:aws:ecr:region:012345678910:repository-namespace/repository-name."})  # fmt: skip
+    registry_id: Optional[str] = field(default=None, metadata={"description": "The Amazon Web Services account ID associated with the registry that contains the repository."})  # fmt: skip
+    repository_uri: Optional[str] = field(default=None, metadata={"description": "The URI for the repository. You can use this URI for container image push and pull operations."})  # fmt: skip
+    image_tag_mutability: Optional[str] = field(default=None, metadata={"description": "The tag mutability setting for the repository."})  # fmt: skip
+    image_scan_on_push: Optional[bool] = field(default=None, metadata={"description": "The image is scanned on every push."})  # fmt: skip
+    encryption_configuration: Optional[AwsEcrEncryptionConfiguration] = field(default=None, metadata={"description": "The encryption configuration for the repository. This determines how the contents of your repository are encrypted at rest."})  # fmt: skip
+
+
+# @define(eq=False, slots=False)
```

Comment:
Want to keep these commented out classes?

Reply:
Yes - not 100% sure if the images are (not) needed.